src/App.tsx

```diff
@@ -14,37 +16,105 @@ export const goodsFromServer = [
   'Jam',
   'Garlic',
 ];
+const REVERSE = false;
+
+enum SortType {
+  RESET_VALUE = '',
+  SORT_BY_NAME = 'name',
+  SORT_BY_LENGTH = 'length',
+}
+
+interface SortCriteria {
+  sortBy: SortType;
+  isReversed: boolean;
+}
+
+function sortGoods(
+  goods: string[],
+  { sortBy: criteria, isReversed }: SortCriteria,
+) {
+  let sortedGoods;
+
+  switch (criteria) {
+    case SortType.SORT_BY_NAME:
+      sortedGoods = goods.toSorted((a, b) => a.localeCompare(b));
+      break;
+    case SortType.SORT_BY_LENGTH:
+      sortedGoods = goods.toSorted((a, b) => a.length - b.length);
+      break;
+
+    default:
+      sortedGoods = [...goods];
+  }
+
+  if (isReversed) {
+    sortedGoods = sortedGoods.toReversed();
+  }
+
+  return sortedGoods;
+}
 
 export const App: React.FC = () => {
+  const [sortBy, setSortBy] = useState(SortType.RESET_VALUE);
+  const [isReversed, setIsReversed] = useState(REVERSE);
+
+  const sortedGoods = sortGoods(goodsFromServer, { sortBy, isReversed });
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={classNames('button', 'is-info', {
+            'is-light': SortType.SORT_BY_NAME !== sortBy,
+          })}
+          onClick={() => setSortBy(SortType.SORT_BY_NAME)}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          type="button"
+          className={classNames('button', 'is-info', {
+            'is-light': SortType.SORT_BY_LENGTH !== sortBy,
+          })}
+          onClick={() => setSortBy(SortType.SORT_BY_LENGTH)}
+        >
           Sort by length
         </button>
 
-        <button type="button" className="button is-warning is-light">
+        <button
+          type="button"
+          className={classNames('button', 'is-info', {
+            'is-light': !isReversed,
+          })}
+          onClick={() => {
+            setIsReversed(!isReversed);
+          }}
+        >
           Reverse
         </button>
 
-        <button type="button" className="button is-danger is-light">
-          Reset
-        </button>
+        {(sortBy || isReversed) && (
+          <button
+            type="button"
+            className="button is-info"
+            onClick={() => {
+              setSortBy(SortType.RESET_VALUE);
+              setIsReversed(REVERSE);
+            }}
+          >
+            Reset
+          </button>
+        )}
       </div>
 
       <ul>
-        <ul>
-          <li data-cy="Good">Dumplings</li>
-          <li data-cy="Good">Carrot</li>
-          <li data-cy="Good">Eggs</li>
-          <li data-cy="Good">Ice cream</li>
-          <li data-cy="Good">Apple</li>
-          <li data-cy="Good">...</li>
-        </ul>
+        {sortedGoods.map(good => (
+          <li data-cy="Good" key={good}>
```

Comment:
[CODE KNOWLEDGE] - When rendering a list, make sure to add a `key` prop that is unique for each element. Using the `good` string itself as a key is risky if the list can have duplicate values. Consider using a combination of the good's name and its index or another unique identifier.